### DISCUSSION TITLE
Missing css/js on custom theme frontpage

### DISCUSSION THREAD

**Discussion by joey-kwl** [Codeinwp/otter-blocks#1359] [Q&A]
### Description:
I'm loading a page with  get_post() combined with apply_filters() into my frontpage of a custom theme, but when I load Otter's slider block all of the images are underneath eachother.

### How to reproduce:

1. Make page with a slider from Otter
2. Load page in frontpage with get_post

### Expected behaviour:
A slider like in the gutenberg

### Current behaviour:
All of the images are shown underneath eachother

### Technical info
* WordPress version: 6.1.1
* Plugin version: 2.1.3

I tried adding this to my frontpage (index.php):
```
use ThemeIsle\GutenbergBlocks\CSS as OtterBlockCSS;
$css = new OtterBlockCSS\Block_Frontend();
$css->enqueue_styles("18", true)
```

Found here: https://wordpress.org/support/topic/otter-block-styles-on-other-page/
but enqueue_styles return NULL

Edit: 
It seems like it went inside this part of class-block-frontend.php:
```
if ( ! has_blocks( $post_id ) ) {
	return;
}
```

**Comment by joey-kwl** [comment]
I found a workaround. Create a Post (not a page but a post) with a slider and you can publish it as private. The plugin will load all the css/js.

**Comment by HardeepAsrani** [comment]
Hey @ChineseOuchie,

Could you send us the custom theme that you've made so we could reproduce on it? Thank you!

**Comment by joey-kwl** [comment]
@HardeepAsrani  I have added you to a temporary repo. If you go into the /dist folder you will see a zip and instructions file

**Comment by HardeepAsrani** [comment]
Hey @ChineseOuchie,

In the index.php file, I used the get_posts function to make a loop like:

```
<div>
	<?php
	global $post;

	$myposts = get_posts( array(
		'posts_per_page' => 5
	) );

	if ( $myposts ) {
		foreach ( $myposts as $post ) : 
			setup_postdata( $post ); ?>
			<?php the_content(); ?>
		<?php
		endforeach;
		wp_reset_postdata();
	}
	?>
</div>
```

And it seems to work properly with the slider. Am I missing something you are doing?

**Reply by joey-kwl** [reply]
Correct me if I'm wrong but you are loading a **Post** while I'm loading a **Page** https://wordpress.com/support/post-vs-page/

```
$about_us = get_theme_mod( "about_us_content", false );
if ($about_us) {
	$about_us_page = get_post(get_theme_mod( "about_us_content", "" ));

					
	$about_us_title = get_theme_mod( "about_us_title", "About us" );;
	$about_us_content = apply_filters("the_content", $about_us_page->post_content);
}
```

```
if ($about_us) {?>
	<div class="about_us mt-5">
		<h1 class="text-uppercase fw-bold mb-4 text-center">
			<?php echo $about_us_title ?>
		</h1>
		<div class="content container">
			<?php echo $about_us_content ?>
		</div>

	</div>
<?php } ?>
```

**Reply by HardeepAsrani** [reply]
I just used it directly with your code and it seems to be working for me, here's a quick demo: https://www.loom.com/share/ee29347608b04ff4a303b5586b1294b6

**Reply by joey-kwl** [reply]
Here is mine demo: https://streamable.com/1ytxza

When I remove the slider from Post, the slider in my Page is gone and all of the images are under eachother.

**Reply by HardeepAsrani** [reply]
Got it, one of the posts probably had a slider which was making it work on my side then. Let me test and let you know.

**Reply by HardeepAsrani** [reply]
You can use this function:
```

function enqueue_otter_style( $id ) {
	if ( ! class_exists( '\ThemeIsle\GutenbergBlocks\CSS\Block_Frontend', false ) ) {
		return;
	}

	$main_instance = ThemeIsle\GutenbergBlocks\Registration::instance();

	if ( ! method_exists( $main_instance, 'enqueue_dependencies' ) ) {
		return;
	}

	$block_frontend_instance = ThemeIsle\GutenbergBlocks\CSS\Block_Frontend::instance();

	if ( ! method_exists( $block_frontend_instance, 'enqueue_styles' ) ) {
		return;
	}

	$main_instance->enqueue_dependencies( $id );
	$block_frontend_instance->enqueue_styles( $id, true );
}
```

And pass the ID where it's being used and it should work. Let me know how it goes!

**Reply by joey-kwl** [reply]
Yes, that works.

Is it possible to do this "automatically", like when using Post? I have some pages in index.php and don't want to check them every time, but for now this is a working solution. Thanks

**Reply by HardeepAsrani** [reply]
@ChineseOuchie You can make a filter out of this and whenever you call for post/page ID, you can pass it through the filter.

**Reply by joey-kwl** [reply]
@HardeepAsrani Sorry to bother you again but how do I apply filters for the page? When using add_filter("the_content") I can't retreive the ID, even with get_the_ID() or the_id().

**Reply by HardeepAsrani** [reply]
For the options that you have, you can have a filter like this:

```
add_filter( 'theme_mod_about_us_content', function( $name ) {
	if ( false === get_theme_mod( $name, false ) ) {
		return $name;
	}

	$page = get_post( get_theme_mod( $name ) );

	if ( null !== $page ) {
		enqueue_otter_style( get_theme_mod( $name ) );
	}

	return $name;
});
```

It's a rough filter that I wrote without actually running it so make sure to test it first.
